### ISSUE TITLE
git_repository_init on existing repo does not repopulate hooks

### ISSUE THREAD

**Issue by csware** [libgit2/libgit2#3343]
[git init](https://git-scm.com/docs/git-init) repopulates the hooks, libgit2 doesn't.

Calling git_repository_init with an empty directory correctly populates the hooks.

**Comment by ethomson** [comment]
No, it does not.  Rather than reusing `init` for this purpose, I would rather see a `git_repository_reinit` or `repopulate_hooks` or something similar.
